site/content/en/references/kustomize/kustomization/patches/_index.md

```diff
@@ -55,122 +51,218 @@ The `name` and `namespace` fields of the patch target selector are
-    allowKindChange: true
+  - patch: |-
+      apiVersion: apps/v1
+      kind: Deployment
+      metadata:
+        name: dummy-app
+        labels:
+          app.kubernetes.io/version: 1.21.0
+  - patch: |-
+      apiVersion: apps/v1
+      kind: Deployment
+      metadata:
+        name: not-used
+      spec:
+        template:
+          spec:
+            containers:
+              - name: nginx
+                image: nginx:1.21.0
+    target:
+      labelSelector: "app.kubernetes.io/name=nginx"
 ```
 
-## Example I
+If a `target` is specified, the `name` contained in the metadata is required but not used.
 
-### Intent
+### Patch using Inline JSON6902
 
-To Make the container image point to a specific version and not to the latest container in the
-registry.
+```yaml
+# kustomization.yaml
+resources:
+- deployment.yaml
+patches:
+  - patch: |-
+      - op: add
+        path: /metadata/labels/app.kubernetes.io~1version
+        value: 1.21.0
+    target:
+      group: apps
+      version: v1
+      kind: Deployment
+  - patch: |-
+      - op: replace
+        path: /spec/template/spec/containers/0/image
+        value: nginx:1.21.0
+    target:
+      labelSelector: "app.kubernetes.io/name=nginx"
+```
+
+The `target` field is always required for JSON6902 patches.  
+A special replacement character `~1` is used to replace `/` in label name.
 
-### File Input
+### Patch using Path Strategic Merge
 
 ```yaml
 # kustomization.yaml
 resources:
 - deployment.yaml
-
 patches:
-- path: patch.yaml
+  - path: add-label.patch.yaml
+  - patch: fix-version.patch.yaml
+    target:
+      labelSelector: "app.kubernetes.io/name=nginx"
 ```
 
+As with the Inline Strategic Merge, the `target` field can be omitted.
+
 ```yaml
-# patch.yaml
+# add-label.patch.yaml
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: the-deployment
-spec:
-  template:
-    containers:
-      - name: the-container
-        image: registry/conatiner:1.0.0
+  name: dummy-app
+  labels:
+    app.kubernetes.io/version: 1.21.0
 ```
 
-### Build Output
-
 ```yaml
+# fix-version.patch.yaml
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: the-deployment
+  name: not-used
 spec:
-  replicas: 5
   template:
-    containers:
-    - image: registry/conatiner:1.0.0
-      name: the-container
+    spec:
+      containers:
+        - name: nginx
+          image: nginx:1.21.0
 ```
 
-## Example II
+As with the Inline Strategic Merge, the `name` field in the patch is not used when a `target` is specified.
 
-### Intent
-
-To Make the container image point to a specific version and not to the latest container in the
-registry.
-
-### File Input
+### Patch using Path JSON6902
 
 ```yaml
 # kustomization.yaml
 resources:
 - deployment.yaml
-
 patches:
-- target:
-    kind: Deployment
-    name: the-deployment
-  path: patch.json
+  - path: add-label.patch.json
+    target:
+      group: apps
+      version: v1
+      kind: Deployment
+  - patch: fix-version.patch.yaml
```

Comment:
just noticed this - is this `patch` here supposed to be `path`?

Reply:
Totally; I indeed modified it when trying locally. Let me double check all examples are working! Thank you